### PR TITLE
Replace Next.js Links with HTML anchor

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

Consistently use server-side navigation links across the codebase.

## Checklist

- [x] Checks pass
- [x] ~Check if changes affect the [resource index JSON revision](https://docs.nextstrain.org/projects/nextstrain-dot-org/en/latest/resource-collection.html#resource-index-revisions)~

### CODE REVIEW

File: static-site/src/components/splash/index.jsx [hunk 1]
```diff
@@ -28,13 +28,33 @@ const Section = ({id, title, abstract, cards, buttonText, buttonLink}) => (
...
-      <Styles.Button to={buttonLink}>
+      <Button to={buttonLink}>
         {buttonText}
-      </Styles.Button>
+      </Button>
     </div>
   </div>
 );
 
+const Button = ({to, children}) => {
```
Comment (genehack) [inline static-site/src/components/splash/index.jsx:38]: would it be better to pull this out into a dedicated component file?

**Review by genehack — COMMENTED** [review]
Wrapping next/link because we _might_ want to switch to plain old HTML in the future has a strong YAGNI vibe, I think.

**Comment by tsibley** [comment]
What @genehack said... I think I'd prefer to either decide _now_ to use `<Link>` or `<a>` (and in general I'd prefer the latter) rather than do a refactor which avoids making that decision.

**Comment by genehack** [comment]
> (and in general I'd prefer the latter) 

FWIW I lean in that direction too, but I noticed from the [next/link docs](https://nextjs.org/docs/app/api-reference/components/link) that `<Link>` does some opportunistic pre-fetching, and I wondered how removing that might affect perceived site responsiveness.

**Comment by tsibley** [comment]
Yeah, it's not just pre-fetching, it's also the difference between client-side navigation and round trips to the server.

**Comment by victorlin** [comment]
I don't have a strong opinion here, but is this part of the broader discussion at https://github.com/nextstrain/private/issues/88?

**Comment by tsibley** [comment]
Sorta, yeah. "Do we want to have client-side navigation between pages? Or is it overkill for nextstrain.org?"

I personally see client-side nav for page-oriented sites like nextstrain.org as effectively an optimization layer, and find that it's often not worth the added complexity/quirks it brings (even when that complexity is wrapped up and tied with a bow by a framework like Next.js).

**Comment by victorlin** [comment]
This should really be discussed on an issue which I should've created before PR, but I think we've enough conversation to continue here rather than starting fresh with a new issue. I also don't want to close this PR or block on https://github.com/nextstrain/private/issues/88 because there's no good reason for the current mix of client and server side navigation.

Given the votes here, I'll rework this PR to replace all client side navigation with server side navigation and drop the wrapper.

File: static-site/src/components/splash/index.jsx (hunk 1 above)
Comment (victorlin) [reply static-site/src/components/splash/index.jsx]: Dropped this change from the PR

**Pull request merged**